### ISSUE TITLE
How does the worm sense the environment in the program? How does `connectome.js` even work?

### ISSUE THREAD

**Issue by multitrack-collector** [heyseth/worm-sim#4]
Hi.

When looking at `connectome.js` I saw four lines starting at 260:

```
/* Use these to stimulate nose and food sensing neurons */
BRAIN.stimulateHungerNeurons = true;
BRAIN.stimulateNoseTouchNeurons = false;
BRAIN.stimulateFoodSenseNeurons = false;
```

Is the worm actually in an endless state of hunger because `BRAIN.stimulateHungerNeurons` is set to `true`? How does it know when it collides with the edges of the window if `BRAIN.stimulateNoseTouchNeurons` is set to `false`? Does `BRAIN.stimulateFoodSenseNeurons` simulate the nematode being "full"?

How exactly does the javascript code work?

Thanks.

**Comment by thomascrha** [comment]
I'm in the same boat - I have no idea how that code works - I'm finding this repo to be more useful and looking at there implementation (which I believe is what this repo is based on)

https://github.com/Connectome/GoPiGo/blob/master/connectome.py

I think this stuff is really out of my wheelhouse - I like computers (they go bleep boop) - but I'm determined to work out whats happening here.

**Comment by heyseth** [comment]
thomascrha is right that you'll have better luck studying the original GoPiGo project, as that is where the implementation for the connectome was originally from. I did not write the JavaScript implementation, I only created the visualization of what the worm's movement would (possibly) look like.

From what I've been able to understand, here's a high level explanation of how the connectome works:

- The BRAIN object is populated with properties that represent the state of the neural network: thresholds for neural activation, accumulators for movement control, and lists of muscles.
- BRAIN.thisState and BRAIN.nextState are used to manage the current and next states of the neural network. This mechanism allows for the simulation of neural activity over time, where the state is updated at each step of the simulation.
- When the accumulated signal at a neuron rises above a certain value, it triggers an action (fires). This simulates the way real neurons only send a signal once a certain threshold of activation is reached.

BRAIN.stimulateHungerNeurons, BRAIN.stimulateNoseTouchNeurons, and BRAIN.stimulateFoodSenseNeurons are just flags which are checked between each state change, and if true, accumulate weighted values on specific neurons and their connected neurons. The BRAIN.postSynaptic dictionary contains the accumulated weighted values as the connectome is executed.
